trieste/acquisition/rule.py

```diff
@@ -1013,6 +1013,11 @@ def __init__(
         self.region_index = region_index
         self.input_active_dims = input_active_dims
 
+    @property
+    def needs_init(self) -> bool:
+        """Return `True` if the search space needs to be initialized, and `False` otherwise."""
+        return False
```

Comment:
Yes, makes sense. I thought about this before, but didn't go ahead with it. Now that you mention it ...

I have made the change, though I kept the "initialization" in `FixedPointTrustRegionDiscrete.__init__`. Safer to set the points to something (rather than `None`), and random init is the most obvious.

Reply:
@uri-granta note though that this change adds a minor incompatibility, explained in the description:

> There is a minor incompatibility that the regions are now required to set `self._initialized` to `True` in their `initialize` method. This instance variable starts off as `False` and should be set to `True` in any derived `initialize` methods.

Thoughts?

Reply:
I think it's fine, though I would document this in the `initialize` abstractmethod docstring. Alternatively, you could make `initialize` not abstract and default to just setting `self._initialized` to `True`, and call it via `super()` everywhere. Though that adds more code for no obvious benifit.

Reply:
Thanks. I have added a comment in the docstring.

I agree that making `initialize` not abstract would be of limited benefit, plus it would not fix the backwards incompatibility, since existing classes would still need to be updated to call `super()...`.